### PR TITLE
Create Products Page

### LINKED ISSUE #56: Products page with filters

### PR DESCRIPTION
**Pull request by KAHMOOSHA**
Created the products page with a search bar and ways to filter products.

### CODE REVIEW

File: client/src/Component/EngineerBanner/style.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    url("./engBanner.jpg");
+  height: 45rem;
+  width: 100%;
+  margin: 0px;
+  background-position: center;
+  background-repeat: no-repeat;
+  background-size: cover;
+}
+
+.engineerDate {
```
Comment (nabilramy) [inline client/src/Component/EngineerBanner/style.css:16]: classes names must be kabab not camel case

File: client/src/Component/Filters/filter.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  outline: none;
+}
+.face__Categories {
+  display: flex;
+  gap: 0.3rem;
+  justify-content: space-around;
+  flex-wrap: wrap;
+}
+
+.face__Categories button {
```
Comment (nabilramy) [inline client/src/Component/Filters/filter.css:25]: here you are using snake!!

File: client/src/Component/Filters/Categories.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Select } from 'antd';
+import { PropTypes } from 'prop-types';
+
+const { Option } = Select;
+
+// eslint-disable-next-line func-names
```
Comment (aaamra) [inline client/src/Component/Filters/Categories.jsx]: remove this line

File: client/src/Pages/productsPage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  const [allCategories, setAllCategories] = useState([]);
+  const [total, setTotal] = useState(); // how many products
+  const [pageSize, setPageSize] = useState(6);
+  const [page, setPage] = useState(2);
+
+  const getAllCategories = () => {
+    axios('/categories').then((res) => {
+      const categoriesList = res.data.data;
+      setAllCategories(categoriesList);
+    });
```
Comment (aaamra) [inline client/src/Pages/productsPage/index.jsx]: use async & await syntax and catch the errors

File: client/src/Pages/productsPage/index.jsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+          },
+        },
+      ).then((res) => {
+        const data = res.data.product.rows;
+        setProducts(data);
+        getAllCategories();
+        getCategorieName(categoryId);
+        setTotal(res.data.product.count);
+      });
+    };
```
Comment (aaamra) [inline client/src/Pages/productsPage/index.jsx]: use async & await syntax and catch the errors

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by mo7amedehab97 — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Review by nabilramy — APPROVED** [review]

**Pull request merged**